### PR TITLE
fix: use tmdb first as metadata provider and fallback to tvdb

### LINKED ISSUE #1137: If a provider lookup fails for an ID, there is no fallback to other providers
### Description

https://github.com/Fallenbagel/jellyseerr/blob/9fa47cbba2a5beff703603861cd163be0a70b100/server/lib/scanners/jellyfin/index.ts#L213-L221

If a show has both IDs assigned but the tvdb does not exist or the lookup fails, the tmdb id is not used to do a fallback lookup, leading to the show not being recognized,

Additionally, since Jellyfin uses TMDB as main provider, I suggest switching the order of lookups to prefer TMDB over TVDB.

### Version

2.1.0

### Steps to Reproduce

1. Add a bad TVDB ID to a show in Jellyfin
2. Add a correct TMDB ID to show in Jellyfin
3. Rescan
4. Show is not recognized

### Screenshots

_No response_

### Logs

_No response_

### Platform

desktop

### Device

Computer

### Operating System

Windows

### Browser

Firefox

### Additional Context

_No response_

### Code of Conduct

- [X] I agree to follow Jellyseerr's Code of Conduct

### PR DESCRIPTION
**Pull request by gauthier-th**
#### Description

This PR changes the order of the metadata provider to TMDB first and then fallback to TheTVDB if no TMDB metadata is available. Previously, TheTVDB was used first and there was no fallback if the latter failed.

#### To-Dos

- [ ] Successful build `pnpm build`
- [ ] Translation keys `pnpm i18n:extract`
- [ ] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #1137

### CODE REVIEW

File: server/lib/scanners/jellyfin/index.ts [hunk 1]
```diff
@@ -210,14 +210,23 @@ class JellyfinScanner {
...
-        tvShow = await this.tmdb.getTvShow({
-          tvId: Number(metadata.ProviderIds.Tmdb),
-        });
+      if (metadata.ProviderIds.Tmdb) {
+        try {
+          tvShow = await this.tmdb.getTvShow({
+            tvId: Number(metadata.ProviderIds.Tmdb),
+          });
+        } catch {
+          /* empty */
```
Comment (Shadowghost) [inline server/lib/scanners/jellyfin/index.ts]: You migth still want to log it

Comment (gauthier-th) [reply server/lib/scanners/jellyfin/index.ts]: Makes sense

Comment (fallenbagel) [reply server/lib/scanners/jellyfin/index.ts]: Debug log

File: server/lib/scanners/jellyfin/index.ts [hunk 2]
```diff
@@ -210,14 +210,25 @@ class JellyfinScanner {
...
+          });
+        }
+      }
+      if (!tvShow && metadata.ProviderIds.Tvdb) {
+        try {
+          tvShow = await this.tmdb.getShowByTvdbId({
+            tvdbId: Number(metadata.ProviderIds.Tvdb),
+          });
+        } catch {
+          /* empty */
```
Comment (Shadowghost) [inline server/lib/scanners/jellyfin/index.ts]: Same here I guess 😅

Comment (gauthier-th) [reply server/lib/scanners/jellyfin/index.ts]: I added it [here](https://github.com/Fallenbagel/jellyseerr/pull/1138/commits/c164d97dde8b4b69891ea482e7b20cdcaaa3a553#diff-799b112b3397343d66cfa5140349c7551e63de0bebbf9bf628b7de831f916691R505-R507)

Comment (gauthier-th) [reply server/lib/scanners/jellyfin/index.ts]: Wait. It's not consistent. For TMDB it's logged only if there is an error while retrieving the tv show from jellyfin, while it's logged if there is a missing ID or and error for TVDB.
Do you think it makes sense to also log when there is no error but just a missing ID or not?

Comment (Shadowghost) [reply server/lib/scanners/jellyfin/index.ts]: I'd say we should log both? Not sure how you guys handle logging though

File: server/lib/scanners/jellyfin/index.ts [hunk 3]
```diff
@@ -491,7 +504,9 @@ class JellyfinScanner {
           }
         });
       } else {
-        this.log(`failed show: ${metadata.Name}`);
+        this.log('No information found for this show', 'debug', {
```
Comment (fallenbagel) [inline server/lib/scanners/jellyfin/index.ts]: Why aren't we adding the `metadata.name`?

**Review by Fallenbagel — APPROVED** [review]

**Comment by devanteweary** [comment]
tldr; Sonarr uses TVDB so let Jellyseerr use it as well.

---

Hey brother. Why not use TVDB first? 
Sonarr uses it exclusively so sometimes there are mismatches with TMDB and TVDB.

For instance, the series called "Dirty Jobs" shows as 11 seasons in TMDB but only 10 in TVDB (they consider one of the seasons a spin-off).

Whether they are correct or not, because Jellyseerr is using TMDB for this series and thinks there is a season 11, but Sonarr only sees 10 seasons, there is a mixmatch. 

I noticed this when someone requested season 11 and it added in Sonarr, but none of the seasons were monitored.  
Because season 11 does not exist in TVDB and therefore Sonarr.

Honestly, I'd love the option to turn TMDB off for series altogether. Or at least choose which one is used first.

Thank you!

**Comment by Fallenbagel** [comment]
@devanteweary this pr is not at all what you think it is. That's not what this pr is for at all.

We don't use TVDB AT ALL. This PR just changes the order at which series are scanned FROM JELLYFIN. To get its external ID. Because we use tmdb. To match with tmdb.
